release notes for 1.30.0 (#17357)
istio/istio.io: content/en/news/releases/1.30.x/announcing-1.30/upgrade-notes/index.md

---
title: Upgrade Notes
description: Important changes to consider when upgrading to Istio 1.30.0.
weight: 20
---

When you upgrade from Istio 1.29.0 to Istio 1.30.0, you need to consider the changes on this page.
These notes detail the changes which purposefully break backwards compatibility with Istio 1.29.x.

> The notes also mention changes which preserve backwards compatibility while introducing new behavior.

Reviewer: Both `changes which` clauses are restrictive — use `that`.

```suggestion
These notes detail the changes that purposefully break backwards compatibility with Istio 1.29.x.
The notes also mention changes that preserve backwards compatibility while introducing new behavior.
```

Changes are only included if the new behavior would be unexpected to a user of Istio 1.29.x.

## CNI config file permissions changed to 0600

The default file permissions for CNI config files written by Istio have changed from 0644 to 0600.
This aligns with the CIS Kubernetes benchmark `v1.12` requirement. Since the CNI config is only read
by the container runtime running as root, this should have no functional impact. If you have tooling
that needs to read CNI config files as a non-root group member, you can set permissions to 0640 by
setting `values.cni.env.CNI_CONF_GROUP_READ=true` environment variable on the
`istio-cni-node` `DaemonSet`.

> ## CNI Agent respects excludeNamespaces configuration

Reviewer 2: ```suggestion
## CNI Agent respects `excludeNamespaces` configuration
```

Previously, only the CNI Plugin respected the `excludeNamespaces` config by skipping the processing of excluded namespace's pods,

> while the CNI Agent would still reconcile and add ambient-labeled Pods in an excluded namespace to the mesh.

Reviewer: Plural possessive (`namespaces'`, since multiple namespaces are excluded) and lowercase `pods` for consistency with the rest of the paragraph.

```suggestion
Previously, only the CNI Plugin respected the `excludeNamespaces` config by skipping the processing of excluded namespaces' pods,
while the CNI Agent would still reconcile and add ambient-labeled pods in an excluded namespace to the mesh.
```

Now, the CNI Agent respects excluded namespaces, which means existing, enrolled pods in an excluded namespace will be un-enrolled, and
new, ambient-labeled pods in an excluded namespace will not be enrolled.

## Untaint controller

> The `PILOT_ENABLE_NODE_UNTAINT_CONTROLLERS` environment variable is now automatically configured when `taint.enabled` is set in the Helm chart when deploying `istiod` chart. Manual activation of this variable in the istiod deployment is no longer necessary.

Reviewer: Double `when` reads awkwardly, plus a missing article before `istiod chart`. Suggesting a rewrite.

```suggestion
The `PILOT_ENABLE_NODE_UNTAINT_CONTROLLERS` environment variable is now automatically configured when `taint.enabled` is set in the Helm chart for the `istiod` deployment. Manual activation of this variable in the `istiod` deployment is no longer necessary.
```

> If you enabled untaint controller using `taint.enabled` Helm value when deploying `istiod` chart, you previously also had to set the `PILOT_ENABLE_NODE_UNTAINT_CONTROLLERS` environment variable in the `istiod` deployment to `true`. This is no longer required, as the variable is now set automatically when the untaint controller is enabled.

Reviewer 3: ```suggestion
The `PILOT_ENABLE_NODE_UNTAINT_CONTROLLERS` environment variable is now automatically configured when `taint.enabled` is set in the Helm chart when deploying `istiod` chart. Manual activation of this variable in the istiod deployment is no longer necessary.
```

I think it can be simpler like this? wdyt?

Author: yes your wording is much better @fjglira - thanks!

## Sidecar proxy service namespace selection changed

When configuring sidecar proxies if a hostname exists in multiple namespaces, Istio now prefers Kubernetes

> services and falls back to the oldest non-Kubernetes service by creation time. Previously, the first visible

Reviewer: Missing comma after the introductory clause, and backtick `Service` (Kubernetes resource).

```suggestion
When configuring sidecar proxies, if a hostname exists in multiple namespaces, Istio now prefers Kubernetes `Service`
resources and falls back to the oldest non-Kubernetes service by creation time. Previously, the first visible
```

namespace alphabetically was chosen.

This may cause traffic to route to a different service instance if you have the same hostname across multiple

> namespaces with mixed service types (e.g. a Kubernetes service and a `ServiceEntry`).

Reviewer: `e.g.` should be followed by a comma, and backtick `Service`.

```suggestion
namespaces with mixed service types (e.g., a Kubernetes `Service` and a `ServiceEntry`).
```
